hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -365,10 +366,12 @@ def incorporate_test_data(self, data):
         """Takes a ConjectureData or Overrun object updates the current
         shrink_target if this data represents an improvement over it,
         returning True if it is."""
-        if data is Overrun or data is self.shrink_target:
+        if data.status < Status.VALID or data is self.shrink_target:
             return
```

Comment:
Unrelated: I just noticed that this method has inconsistent return types (`None` vs `True`/`False`).

(We get away with it because the only caller currently doesn't even use the return value, and `None` is falsy anyway.)

Reply:
Good point. I might as well fix that while I'm here.